crates/re_ui/src/list_item2/property_content.rs

```diff
@@ -0,0 +1,278 @@
+    where
+        F: FnOnce(&ReUi, &mut egui::Ui, egui::style::WidgetVisuals) -> Option<egui::Response> + 'a,
+    {
+        self.value_fn = Some(Box::new(value_fn));
+        self
+    }
+
+    //
+    // Bunch of helpers with concrete implementation of value fn
+    //
+
+    /// Show a read-only boolean in the value column.
+    #[inline]
+    pub fn value_bool(self, mut b: bool) -> Self {
+        self.value_fn(move |_, ui: &mut Ui, _| {
+            Some(ui.add_enabled(false, crate::toggle_switch(15.0, &mut b)))
+        })
+    }
+
+    /// Show an editable boolean in the value column.
+    #[inline]
+    pub fn value_bool_mut(self, b: &'a mut bool) -> Self {
+        self.value_fn(|_, ui: &mut Ui, _| {
+            ui.visuals_mut().widgets.hovered.expansion = 0.0;
+            ui.visuals_mut().widgets.active.expansion = 0.0;
+
+            Some(ui.add(crate::toggle_switch(15.0, b)))
+        })
+    }
+
+    /// Show a static text in the value column.
+    #[inline]
+    pub fn value_text(self, text: impl Into<egui::WidgetText> + 'a) -> Self {
+        self.value_fn(move |_, ui, _| Some(ui.label(text.into())))
+    }
+
+    /// Show an editable text in the value column.
+    #[inline]
+    pub fn value_text_mut(self, text: &'a mut String) -> Self {
+        self.value_fn(|_, ui, _| Some(ui.text_edit_singleline(text)))
+    }
+
+    /// Show a read-only color in the value column.
+    #[inline]
+    pub fn value_color(self, color: &'a [u8; 4]) -> Self {
+        self.value_fn(|_, ui, _| {
+            let [r, g, b, a] = color;
+            let color = egui::Color32::from_rgba_unmultiplied(*r, *g, *b, *a);
+            let response = egui::color_picker::show_color(ui, color, ui.spacing().interact_size);
+            response.on_hover_text(format!("Color #{r:02x}{g:02x}{b:02x}{a:02x}"));
+            None
+        })
+    }
+
+    /// Show an editable color in the value column.
+    #[inline]
+    pub fn value_color_mut(self, color: &'a mut [u8; 4]) -> Self {
+        self.value_fn(|_, ui: &mut egui::Ui, _| {
+            ui.visuals_mut().widgets.hovered.expansion = 0.0;
+            ui.visuals_mut().widgets.active.expansion = 0.0;
+            Some(ui.color_edit_button_srgba_unmultiplied(color))
+        })
+    }
+}
+
+impl ListItemContent for PropertyContent<'_> {
+    fn ui(
+        self: Box<Self>,
+        re_ui: &ReUi,
+        ui: &mut Ui,
+        context: &ContentContext<'_>,
+    ) -> Option<Response> {
+        let Self {
+            label,
+            icon_fn,
+            summary_only,
+            value_fn,
+            action_buttons,
+        } = *self;
+
+        // We always reserve space for the action button(s), even if there are none.
+        let action_button_rect = egui::Rect::from_center_size(
+            context.rect.right_center() - egui::vec2(ReUi::small_icon_size().x / 2., 0.0),
+            ReUi::small_icon_size() + egui::vec2(1.0, 1.0), // padding is needed for the buttons
```

Comment:
This padding is not allocated on the right side though, so I think there is an off-by-padding error here.

An ASCII diagram is needed I think.

Reply:
Also, why is the padding exactly 1 pixel?_why_ is it needed?

Reply:
Yeah this was really hack. I've cleaned up that code. Not only it's less obtuse now, its actually better aligned indeed.